src/vs/workbench/contrib/files/browser/views/explorerView.ts

```diff
@@ -176,7 +176,13 @@ export class ExplorerView extends ViewletPanel {
 			const isEditing = !!this.explorerService.getEditableData(e);
 
 			if (isEditing) {
-				await this.tree.expand(e.parent!);
+				// Don't expand when parent is the root in single folder
+				// Fixes https://github.com/Microsoft/vscode/issues/72626
+				const isSingleFolder = this.contextService.getWorkbenchState() === WorkbenchState.FOLDER;
```

Comment:
Use this.tree.getInput() to check if the e.parent is shown or not.
Also use e.parent.isRoot()
Try not to use the exclamation mark to fool the ts compiler.